nerfstudio/models/splatfacto.py

```diff
@@ -757,16 +783,27 @@ def get_outputs(self, camera: Cameras) -> Dict[str, Union[torch.Tensor, List]]:
             info["means2d"].retain_grad()
         self.xys = info["means2d"]  # [1, N, 2]
         self.radii = info["radii"][0]  # [N]
-
         alpha = alpha[:, ...]
+
+        background = self._get_background_color()
         rgb = render[:, ..., :3] + (1 - alpha) * background
         rgb = torch.clamp(rgb, 0.0, 1.0)
+
         if render_mode == "RGB+ED":
             depth_im = render[:, ..., 3:4]
             depth_im = torch.where(alpha > 0, depth_im, depth_im.detach().max()).squeeze(0)
         else:
             depth_im = None
-        return {"rgb": rgb.squeeze(0), "depth": depth_im, "accumulation": alpha.squeeze(0), "background": background}  # type: ignore
+
+        if background.shape[0] == 3:
+            background = background.expand(H, W, 3)
+
+        return {
+            "rgb": rgb.squeeze(0),  # type: ignore
+            "depth": depth_im,  # type: ignore
+            "accumulation": alpha.squeeze(0),  # type: ignore
+            "background": background.squeeze(0),  # type: ignore
```

Comment:
I don't understand if you squeeze(0) for tensor shape in (H, W, 3), what would you expect?

Also can you also validate if the current code allow alpha transparency training on blender dataset. See [this](https://github.com/nerfstudio-project/nerfstudio/pull/3088) for instructions.

Reply:
I copied this part from my splatw pr and i forget to clean it all up(

Reply:
Yes, transparency training on blender dataset still works